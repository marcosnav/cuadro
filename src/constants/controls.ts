/**
 * Available control types
 */
export enum ControlType {
  NEW_GAME = 'NEW_GAME',
  RESTART_GAME = 'RESTART_GAME',
  SEE_ORIGINAL = 'SEE_ORIGINAL',
}
