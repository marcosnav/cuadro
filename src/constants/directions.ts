/**
 * SwipeDirection, one of UP, RIGHT, LEFT, DOWN
 */
export enum SwipeDirection {
  UP = 'UP',
  RIGHT = 'RIGHT',
  DOWN = 'DOWN',
  LEFT = 'LEFT',
}
