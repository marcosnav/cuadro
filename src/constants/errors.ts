export enum errors {
  NO_INTERNET = 'NO_INTERNET',
  UNEXPECTED_ERROR = 'UNEXPECTED_ERROR',
  UNRESPONSIVE_IMAGE_PROVIDER = 'UNRESPONSIVE_IMAGE_PROVIDER',
}
