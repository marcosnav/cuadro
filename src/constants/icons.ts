import check from './../assets/svg/check.svg';
import image from './../assets/svg/image.svg';
import newicon from './../assets/svg/new.svg';
import refresh from './../assets/svg/refresh.svg';

export const icons: { [key: string]: string } = {
  check,
  image,
  new: newicon,
  refresh,
};
