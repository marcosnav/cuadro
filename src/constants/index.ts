export * from './status';
export * from './errors';
export * from './theme';
export * from './keycodes';
