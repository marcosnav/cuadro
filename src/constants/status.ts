export enum status {
  LOADING_IMAGE = 'LOADING_IMAGE',
  DISPLAY_ORIGINAL = 'DISPLAY_ORIGINAL',
  PLAYING_GAME  = 'PLAYING_GAME ',
  FINISHED_GAME = 'FINISHED_GAME',
}
