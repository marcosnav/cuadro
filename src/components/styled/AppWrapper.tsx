import styled, { StyledProps } from 'styled-components';
import { Status } from './../../constants';

interface IProps {
  status?: Status;
}

const getGradient = (props: StyledProps<IProps>) => {
  const startingGame = props.status === Status.STARTING_NEW_GAME;
  const loadingImage = props.status === Status.LOADING_IMAGE;
  const errored = props.status === Status.ERROR;

  let startG = props.theme.GD_PURPLE;
  let endG = props.theme.GD_BLUE;

  if (startingGame || loadingImage) {
    startG = props.theme.GD_DARK_PURPLE;
    endG = props.theme.GD_PURPLE;
  }

  if (errored) {
    startG = props.theme.GD_DARK_RED;
    endG = props.theme.RED;
  }

  return `linear-gradient(180deg, ${startG} 0%, ${endG} 100%)`;
};

const AppWrapper = styled.div<IProps>`
  align-items: center;
  background: ${(props) => props.theme.PURPLE};
  background: ${ (props) => getGradient(props) };
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  height: 100vh;
  left: 0;
  padding: 26px;
  position: fixed;
  top: 0;
  transition: background 0.3s ease;
  width: 100%;
`;

AppWrapper.displayName = 'AppWrapper';

export default AppWrapper;
