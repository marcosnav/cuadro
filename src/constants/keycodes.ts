import { SwipeDirection } from './directions';

/**
 * KeyCodes to match for keypress events
 */
export enum KeyCodes {
  DOWN = 40,
  LEFT = 37,
  RIGHT = 39,
  UP = 38,
  ENTER = 13,
}

/**
 * Map key code with swipe direction
 */
export const KeySwipeMap: { [key: number]: SwipeDirection } = {
  40: SwipeDirection.DOWN,
  37: SwipeDirection.LEFT,
  39: SwipeDirection.RIGHT,
  38: SwipeDirection.UP,
};
