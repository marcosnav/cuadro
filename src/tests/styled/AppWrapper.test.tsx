import { mount } from 'enzyme';
import 'jest-styled-components';
import React from 'react';
import { ThemeConsumer } from 'styled-components';
import AppWrapper from './../../components/styled/AppWrapper';
import { Status, theme } from './../../constants';

describe('<AppWrapper/>', () => {
  beforeAll(() => {
    ThemeConsumer._currentValue = theme;
  });

  test('renders properly', () => {
    const component = mount(<AppWrapper />);
    expect(component).toHaveStyleRule(
      'background',
      `linear-gradient(180deg,${theme.GD_PURPLE} 0%,${theme.GD_BLUE} 100%)`,
    );
    expect(component).toHaveStyleRule('position', 'fixed');
    expect(component).toHaveStyleRule('height', '100vh');
    expect(component).toHaveStyleRule('width', '100%');
    expect(component).toMatchSnapshot();
  });

  describe('on another app states', () => {
    test('on LOADING', () => {
      const component = mount(<AppWrapper status={Status.LOADING_IMAGE} />);
      expect(component).toHaveStyleRule(
        'background', 
        `linear-gradient(180deg,${theme.GD_DARK_PURPLE} 0%,${theme.GD_PURPLE} 100%)`,
      );
    });

    test('on ERROR', () => {
      const component = mount(<AppWrapper status={Status.ERROR} />);
      expect(component).toHaveStyleRule(
        'background', 
        `linear-gradient(180deg,${theme.GD_DARK_RED} 0%,${theme.RED} 100%)`,
      );
    });
  });
});
