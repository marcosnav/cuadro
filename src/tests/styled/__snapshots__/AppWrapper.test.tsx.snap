// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppWrapper/> renders properly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: linear-gradient(180deg,#7D05AA 0%,#238BFF 100%);
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  padding: 26px;
  position: fixed;
  width: 100%;
}

<AppWrapper>
  <div
    className="c0"
  />
</AppWrapper>
`;
